charts/metal3/0.2.1/charts/baremetal-operator/values.yaml

```diff
@@ -0,0 +1,101 @@
+# Default values for baremetal-operator.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+global:
+  # specify comma serparate beginning and end of the range of IP
+  # addresses the DHCP server will manage.
+  dhcpRange: 192.168.20.20,192.168.20.80
+
+  # Network interface on which provisioning network can be accessed
+  provisioningInterface: ens4
+
+  # IP Address assigned to network interface on provisioning network
+  provisioningIP: 192.168.20.5
+
+  # If running in a multi-node kubernetes cluster, "pin" the baremtal container
+  # to the same host where the ironic and media containers
+  # arerunning. Uncomment the nodeSelector and update the hostname accordingly.
+  #nodeSelector: 
+    #kubernetes.io/hostname: "my-hostname"
+
+  # Comment this out when pinning the baremetal-operator container to a specfic host.
+  nodeSelector: {}
+
+replicaCount: 1
+
+image:
+  repository: registry.opensuse.org/isv/metal3/bci/baremetal-operator/containerfile/suse/baremetal-operator
+  pullPolicy: IfNotPresent 
+  tag: "0.4.0"
+
+imagePullSecrets: []
+nameOverride: "manger"
+fullnameOverride: "baremetal-operator"
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
+  name: "baremetal-operator-controller-manager"
+
+podAnnotations: {}
+
+securityContext:
+  runAsUser: 11000
```

Comment:
As the `imagePullPolicy` in our charts is `IfNotPresent`, this will not gonna work for existing clusters. Probably the imagePullPolicy could be switched to `Always` for the current release and reverted back in the future...

Reply:
IMO it is not a very good idea to always pull images. This is a tasking task and might cause additional problems such as laggy internet connection causing errors and so on. 

My 2c for the Sylva setup, I think we should keep this as `IfNotPresent` and in the `metal3-suse` unit in Sylva provide the image pull policy. Having a default `Always` pull policy will cause more problems than solutions.

Reply:
In general I agree `Always` isn't the best solution,  but because OBS can build new image revisions without updating the tag (and can't maintain older tags without branching the package) I think it's reasonable for the chart defaults.

We can certainly modify that default for Sylva deployments in the values, and would likely want `IfNotPresent` in future downstream chart artefacts where we can rely on different image-tagging behavior.